File: infullmvp/src/main/java/com/infullmobile/android/infullmvp/PresentedView.java
package com.infullmobile.android.infullmvp;

import android.content.Context;
import android.support.annotation.IdRes;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.view.View;

public abstract class PresentedView<PresenterType extends Presenter, PresentedViewType> {

    protected PresenterType boundPresenter;

    @LayoutRes
    protected abstract int getLayoutResId();
    protected abstract void bindUiElements(@NonNull PresentedViewType boundingView, @NonNull PresenterType presenter);
    protected abstract void bindViews();
    protected abstract void onViewsBound();
    protected abstract <ViewType extends View> ViewType findView(@IdRes int resourceId);

    @NonNull
    public abstract Context getContext();

    public void bindPresenter(@NonNull PresenterType presenter) {
        boundPresenter = presenter;
    }

    @NonNull
    public PresenterType getBoundPresenter() {
        return boundPresenter;
    }
}

File: infullmvp-kotlin-koin/src/main/java/com/infullmobile/android/infullmvp/PresentedView.kt
package com.infullmobile.android.infullmvp

import kotlinx.android.extensions.LayoutContainer

abstract class PresentedView<PresenterType : InFullMvpPresenter<*>> : MvpView, LayoutContainer {

    lateinit var presenter: PresenterType

    fun bindView(presenter: PresenterType) {
        this.presenter = presenter
        onViewBound()
    }

    abstract val scopeName: String
    abstract val layoutId: Int
}